### PR TITLE
Implemented request pool keeper

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed: #XXXX

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: x/oracle/types/keys.go [hunk 1]
```diff
@@ -1,6 +1,9 @@
 package types
 
 import (
+	"crypto/sha256"
+	fmt "fmt"
```
Comment (taobun) [inline x/oracle/types/keys.go:5]: fmt?

File: x/oracle/types/keys.go [hunk 2]
```diff
@@ -110,3 +113,19 @@ func ReportsOfValidatorPrefixKey(reqID RequestID, val sdk.ValAddress) []byte {
 func ReportersOfValidatorPrefixKey(val sdk.ValAddress) []byte {
 	return append(ReporterStoreKeyPrefix, val.Bytes()...)
 }
+
+// GetEscrowAddress returns the escrow address for the specified channel.
```
Comment (taobun) [inline x/oracle/types/keys.go:117]: and request key

File: x/oracle/types/keys.go [hunk 3]
```diff
@@ -110,3 +113,19 @@ func ReportsOfValidatorPrefixKey(reqID RequestID, val sdk.ValAddress) []byte {
...
+// The escrow address follows the format as outlined in ADR 028:
+// https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-028-public-key-addresses.md
+func GetEscrowAddress(requestKey, portID, channelID string) sdk.AccAddress {
+	// a slash is used to create domain separation between port and channel identifiers to
```
Comment (taobun) [inline x/oracle/types/keys.go:121]: update comment

File: x/oracle/testapp/setup.go [hunk 4]
```diff
@@ -48,9 +48,16 @@ var (
 
 // nolint
 var (
+	Coins1unband        = sdk.NewCoins(sdk.NewInt64Coin("uband", 1))
```
Comment (taobun) [inline x/oracle/testapp/setup.go:51]: unband?

File: x/oracle/keeper/request_pool.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package keeper
+
+import (
+	"github.com/bandprotocol/chain/x/oracle/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) DepositRequestPool(ctx sdk.Context, requestKey string, portID string, channelID string, amount sdk.Coins, sender sdk.AccAddress) error {
```
Comment (taobun) [inline x/oracle/keeper/request_pool.go:8]: Move to internal function for test only.

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

File: x/oracle/types/keys.go [hunk 6]
```diff
@@ -110,3 +113,17 @@ func ReportsOfValidatorPrefixKey(reqID RequestID, val sdk.ValAddress) []byte {
...
 }
+
+// GetEscrowAddress returns the escrow address for the specified channel and request key.
+// The escrow address follows the format as outlined in ADR 028:
+// https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-028-public-key-addresses.md
+func GetEscrowAddress(requestKey, portID, channelID string) sdk.AccAddress {
+	contents := fmt.Sprintf("%s/%s/%s", requestKey, portID, channelID)
+
+	// ADR 028 AddressHash construction
+	preImage := []byte(Version)
```
Comment (ntchjb) [inline x/oracle/types/keys.go:124]: Should we use `func Hash(typ string, key []byte) []byte` instead where `typ` is `Version`?

Comment (ntchjb) [reply x/oracle/types/keys.go:124]: Oh, the function will be available in the future.

https://github.com/cosmos/cosmos-sdk/pull/8415

May use the current for now.

**Review by ntchjb — APPROVED** [review]

**Pull request merged**